fulp_modules/features/antagonists/infiltrators/infil_objectives.dm

```diff
@@ -0,0 +1,204 @@
+	for(var/mob/living/carbon/human/player in GLOB.player_list)
+		if(player.stat == DEAD || player.mind == owner)
+			continue
+		if(player.mind?.has_antag_datum(/datum/antagonist/traitor))
+			possible_targets += player.mind
+
+	if(!possible_targets.len)
+		find_target() //if no traitors on station, this becomes a normal assassination obj
+		return
+	else
+		target = pick(possible_targets)
+
+	if(target?.current)
+		explanation_text = "Special intel has identified [target.name] the [!target_role_type ? target.assigned_role.title : target.special_role]. as a threat to Nanotrasen. Eliminate them at all costs."
+
+
+//advanced mulligan objective
+
+/datum/objective/escape/escape_with_identity/infiltrator
+	name = "escape with identity (as infiltrator)"
+
+/datum/objective/escape/escape_with_identity/infiltrator/proc/find_sec_target()
+	var/list/sec = SSjob.get_all_sec()
+	if(!sec.len)
+		find_target()
+	else
+		target = pick(sec)
+	update_explanation_text()
+
+//Animal Rights Consortium Objectives
+
+//pet killing
+
+/datum/objective/kill_pet
+	var/mob/living/target_pet
+	martyr_compatible = TRUE
+
+/datum/objective/kill_pet/proc/find_pet_target()
+	var/list/possible_target_pets = list (/mob/living/simple_animal/pet/dog/corgi/ian,
+	/mob/living/simple_animal/pet/fox/renault, /mob/living/simple_animal/pet/cat/runtime,
+	/mob/living/simple_animal/parrot/poly)
+
+	remove_duplicate(possible_target_pets) //removes pets from the list that are already in the owner's objective
+
+	var/chosen_pet = pick(possible_target_pets)
+	target_pet = locate(chosen_pet) in GLOB.mob_living_list
+
+	update_explanation_text()
+
+/datum/objective/kill_pet/proc/remove_duplicate(possible_target_pets)
+	for(var/datum/objective/kill_pet/objective in owner.get_all_objectives())
+		if(objective.target_pet.type in possible_target_pets)
+			possible_target_pets -= objective.target_pet.type
+
+
+/datum/objective/kill_pet/update_explanation_text()
+	..()
+	if(target_pet)
+		explanation_text = "[target_pet] has been tainted by Nanotrasen agenda, give them a mercy killing."
+
+
+/datum/objective/kill_pet/check_completion()
+	return completed || (target_pet.stat == DEAD)
+
+//scientist killing
+
+/datum/objective/assassinate/proc/find_sci_target()
+	var/list/sci_targets = list()
+	for(var/mob/living/carbon/human/player as anything in GLOB.human_list)
+		if(player.stat == DEAD)
+			continue
+		if((player.mind?.assigned_role.departments_bitflags & DEPARTMENT_BITFLAG_SCIENCE))
+			sci_targets += player.mind
+
+	if(!sci_targets.len)
+		find_target()
+		return
+	else
+		target = pick(sci_targets)
+
+	if(target?.current)
+		explanation_text = "Make a stance against science's animal experimentation by assassinating [target.name] the [!target_role_type ? target.assigned_role.title : target.special_role]!"
+
+//monkifying head of staff
+
+/datum/objective/assassinate/monkify
+
+/datum/objective/assassinate/monkify/proc/find_head_target()
+	var/list/com_targets = SSjob.get_all_heads()
```

Comment:
Is this taking into account gene-less species? like beefmen/plasmamen?

Reply:
good poin